components/setup/DpuSetup.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import type { Session } from "next-auth";
+
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { MdContentCopy } from "react-icons/md";
+import { getUserRepositories } from '../../utils/providerAPI/getUserRepositories';
+import Button from '../Button';
+
+interface DpuSetupProps {
+	userId: string;
+	selectedProvider: string;
+	selectedInstallationType: string;
+	session: Session | null;
+}
+
+interface SaveSetupReposApiBodyArgs {
```

Comment:
Classes and interfaces should not start with a verb; the verb indicates that this entity will perform an action. It would be better to rename this interface to `SetupReposApiBodyArgs`.